### ISSUE TITLE
wheel scrolling doesn't work in tmux / linux

### ISSUE THREAD

**Issue by Canop** [Canop/bacon#174]
Reported by @willhansen in #153 and confirmed.

**Comment by Canop** [comment]
Related: https://github.com/crossterm-rs/crossterm/issues/401
(bacon uses crossterm for event detection)

**Comment by jyn514** [comment]
fwiw this works around the issue:
```tmux
bind-key -n WheelUpPane {
  if-shell -F '#{mouse_any_flag}' {
      send-keys -M
  } {
    if-shell -F '#{alternate_on}' {
      send-keys Up
    } {
      copy-mode -e
    }
  }
}
bind-key -n WheelDownPane {
  if-shell -F '#{mouse_any_flag}' {
      send-keys -M
  } {
      send-keys Down
  }
}
```
(see also https://github.com/tmux/tmux/issues/3705)

you can duplicate `send-keys` more times to increase the scroll speed.

**Comment by jyn514** [comment]
strangely scrolling works fine in vim and helix. i wonder if crossterm isn't telling the terminal emulator it can accept mouse events (`mouse_any_flag`). i notice https://github.com/Canop/bacon/commit/ead2e058f0b71b88993716c5e3db760e3f7a7cc9 only enables mouse events on windows - do you know why that was done?

**Comment by jyn514** [comment]
ok yes - if i uncomment those cfgs, mouse scrolling works fine in tmux. the problem from https://github.com/Canop/bacon/issues/191 also appears; i suspect it is not actually platform specific and bacon needs to set some config that tells the terminal emulator to handle text selection itself.

**Comment by jyn514** [comment]
ok, i think i have figured out what's going on.

1. by default, the terminal emulator does not send any mouse events to the active process (i.e. the one with the "controlling terminal"); instead it scrolls up in the history.
2. when the "alternate screen buffer" is active, the terminal emulator turns mouse scroll events into up and down arrow keys: https://stackoverflow.com/a/71373998. this happens *even if* the process has not requested to be in mouse mode. click + drag is still processed by the terminal emulator. this is the "happy path" for linux.
3. processes can request to be in mouse mode, which is what https://github.com/Canop/bacon/pull/156 enabled on Windows. in this mode *both* scroll events and click and drag are passed through to the process. this is why https://github.com/Canop/bacon/issues/191 is broken; the terminal emulator expects the events to be handled by bacon.

there are *two* independent bugs here, excluding anything bacon does.
- tmux handles case 2 incorrectly, by not translating scroll events into arrow keys. this is what i fixed above in https://github.com/Canop/bacon/issues/174#issuecomment-2290165179; i am happy to make a PR upstream to tmux with that fix. 
- whatever was going on in #153. something weird is going on, because i tested the following cases with `tput smcup`, all of which worked correctly (translated mouse scroll events to up arrows):
  - git bash
  - WSL2 in Windows terminal
  - WSL2 in cmd.exe (in windows terminal)

i suspect @Adham-A has some broken environment.

given all that, my suggestion is to revert #156, tell tmux users to use the workaround i posted above, and have Adham debug their environment further - i do not think capturing the mouse is a good tradeoff for bacon since it breaks click+drag for selecting text.

**Comment by jyn514** [comment]
> whatever was going on in #153

oh - i think this is a bug in vscode. running `tput smcup` in its integrated terminal fails to translate scroll events to up and down keys. i'm not sure exactly the context there, https://github.com/microsoft/terminal/issues/13187, *seems* to say this should be supported, but it's clearly not. in any case i don't think bacon should try to work around bugs in terminal emulators.

**Comment by MolotovCherry** [comment]
> given all that, my suggestion is to revert [#156](https://github.com/Canop/bacon/pull/156), tell tmux users to use the workaround i posted above, and have Adham debug their environment further - i do not think capturing the mouse is a good tradeoff for bacon since it breaks click+drag for selecting text.

I'm on Windows; I reverted this in my local copy and rebuilt, and I can confirm my scroll works perfectly fine, and I can finally select text again. (Using Windows Terminal + nushell)

**Comment by Canop** [comment]
Just a note regarding

> i do not think capturing the mouse is a good tradeoff for bacon since it breaks click+drag for selecting text.

selecting text with click+drag is broken anyway in bacon due to bacon managing wrapping by adding a scrollbar which then breaks any multi-line selection.

**Comment by jyn514** [comment]
> Just a note regarding
> 
> > i do not think capturing the mouse is a good tradeoff for bacon since it breaks click+drag for selecting text.
> 
> selecting text with click+drag is broken anyway in bacon due to bacon managing wrapping by adding a scrollbar which then breaks any multi-line selection.

yes. but selecting within a line still works on linux, but is broken on windows, because of https://github.com/Canop/bacon/pull/156.
